src/ng/directive/input.js

```diff
@@ -1126,7 +1128,11 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     });
   }
 
-  var timeout;
+  var timeout, oldVal;
+  var viewValueUpdated = false, msieInput = msie >= 10 && msie <= 11;
+  if (msieInput && attr.type in IE_INPUTS_WITH_CLEARING) {
```

Comment:
If we keep it I think it should be part of the `msieInput` condition, not this setup condition.

But I'd think `element.val()` is cheap enough it doesn't matter.